### PR TITLE
Expose function to style buttons with svg icon

### PR DESCRIPTION
**Pull request by sahirboghani**

### CODE REVIEW

File: src/launchAsync.ts [hunk 1]
```diff
@@ -126,3 +126,18 @@ export function launchAsync(token: string, content: Content, options?: Options):
         document.documentElement.style.overflow = 'hidden';
     });
 }
+
+/**
+ * Style the launch buttons with an encoded SVG of the assets\icon.svg
+ */
+export function styleLaunchButtons(): void {
```
Comment (metanMSFT) [inline src/launchAsync.ts]: Add enum for the type of button (e.g. image only, image with text, text only). For the "Immersive Reader" string, might make sense to hardcode all the localized strings in the TS somewhere.

File: src/launchAsync.ts [hunk 2]
```diff
@@ -126,3 +126,18 @@ export function launchAsync(token: string, content: Content, options?: Options):
...
+    const encodedSvg = `url("data:image/svg+xml,%3Csvg viewBox='0 0 40 37' version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'%3E%3Cg stroke='none' stroke-width='1' fill='none' fill-rule='evenodd'%3E%3Cg fill-rule='nonzero'%3E%3Cpath d='M37.4,0.9 L37.4,9.6 L35.4,9.6 L35.4,2.9 L24.4,2.9 C22.9,3.3 20,4.5 20,6 L20,17.2 L18,17.2 L18,6 C18,5 15.6,3.6 13.8,2.9 L2,2.9 L2,29 L12.4,29 L12.4,31 L0,31 L0,0.9 L14.1,0.9 L14.3,1 C15,1.2 17.5,2.2 18.9,3.7 C20.5,1.9 23.5,1.1 23.9,1 L24.1,1 L37.4,1 L37.4,0.9 Z' fill='%23000000'%3E%3C/path%3E%3Cpath d='M27.4,37 L25.8,37 L18.4,29.4 L14,29.4 L14,21 L18.4,20.9 L26.1,13 L27.4,13 L27.4,37 Z M16,27.4 L19.2,27.4 L25.3,33.7 L25.3,16.6 L19.2,22.9 L15.9,22.9 L15.9,27.4 L16,27.4 Z' fill='%230197F2'%3E%3C/path%3E%3Cpath d='M31.3,32.7 L29.6,31.7 C29.6,31.7 31.7,28.3 31.7,25.2 C31.7,21.9 29.6,18.5 29.6,18.4 L31.3,17.4 C31.4,17.6 33.7,21.3 33.7,25.2 C33.7,28.8 31.4,32.6 31.3,32.7 Z' fill='%230197F2'%3E%3C/path%3E%3Cpath d='M36.4,36.2 L34.8,35 C34.8,35 38,30.8 38,25.2 C38,19.6 34.8,15.4 34.8,15.4 L36.4,14.2 C36.5,14.4 40,19 40,25.3 C40,31.5 36.5,36 36.4,36.2 Z' fill='%230197F2'%3E%3C/path%3E%3C/g%3E%3C/g%3E%3C/svg%3E")`;
```
Comment (metanMSFT) [inline src/launchAsync.ts]: Backticks?

File: src/launchAsync.ts [hunk 3]
```diff
@@ -126,3 +126,18 @@ export function launchAsync(token: string, content: Content, options?: Options):
         document.documentElement.style.overflow = 'hidden';
     });
 }
+
+/**
+ * Style the launch buttons with an encoded SVG of the assets\icon.svg
+ */
+export function styleLaunchButtons(): void {
+    const buttons: HTMLCollectionOf<Element> = document.getElementsByClassName('IRLaunchButton');
```
Comment (metanMSFT) [inline src/launchAsync.ts]: Probably good to let the classname be a parameter

File: src/launchAsync.ts [hunk 4]
```diff
@@ -126,3 +126,18 @@ export function launchAsync(token: string, content: Content, options?: Options):
         document.documentElement.style.overflow = 'hidden';
     });
 }
+
+/**
+ * Style the launch buttons with an encoded SVG of the assets\icon.svg
```
Comment (metanMSFT) [inline src/launchAsync.ts]: Maybe describe what the function does from the perspective of the user

File: src/launchAsync.ts (hunk 1 above)
Comment (sahirboghani) [reply src/launchAsync.ts]: Will add that extended functionality later

File: src/launchAsync.ts [hunk 5]
```diff
@@ -134,3 +134,21 @@ export function launchAsync(token: string, content: Content, options?: Options):
         document.documentElement.style.overflow = 'hidden';
     });
 }
+
+/**
+ * Append an <img/> of assets/icon.svg to every element with the specified className. Intended to be used with divs to avoid default styles
+ */
+export function styleLaunchDivs(className: string): void {
+    const launchDivs: HTMLCollectionOf<Element> = document.getElementsByClassName(className);
+    const iconImagePath = '../../assets/icon.svg';
```
Comment (metanMSFT) [inline src/launchAsync.ts]: This should be an absolute path to the CDN

File: src/launchAsync.ts [hunk 6]
```diff
@@ -1,136 +1,154 @@
...
+        // Disable body scrolling
+        document.body.style.overflow = 'hidden';
+        document.documentElement.style.overflow = 'hidden';
+    });
+}
+
+/**
+ * Append an <img/> of assets/icon.svg to every element with the specified className. Intended to be used with divs to avoid default styles
+ */
+export function styleLaunchDivs(className: string): void {
```
Comment (metanMSFT) [inline src/launchAsync.ts]: How about `renderLaunchButtons`?

File: tests/launchAsync.test.ts [hunk 7]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-import { launchAsync } from '../src/launchAsync';
+import { launchAsync, styleLaunchDivs } from '../src/launchAsync';
```
Comment (metanMSFT) [inline tests/launchAsync.test.ts]: Rename

**Pull request merged**